src/FreeAge/client/render_window.cpp

```diff
@@ -4121,11 +4219,95 @@ void RenderWindow::RightClickOnMap(bool haveScreenCoord, const QPoint& screenCoo
   }
 }
 
+bool RenderWindow::SetActiveCommandButton(CommandButton* commandButton, bool allowToggle) {
```

Comment:
I'm trying to add some structure to the "ActiveCommandButton" in order to be used in a unified way by the current `Contruction`, `Garrison`  and the future `Repair`, `AttackMove`, `Convert` and whatever else needs more information that a simple click.